### PR TITLE
feat: implement compatibility with nixos and musl via features and lz…

### PR DESCRIPTION
**Pull request by afsec**
This enhancement aims to broaden the reach of the project by ensuring seamless integration with NixOS and Musl-based systems while maintaining performance and reliability.

### CODE REVIEW

**Comment by afsec** [comment]
I have submitted this PR with the aim of enhancing the portability of parity-db across various architectures. Your review and feedback on this improvement would be greatly appreciated.

**Comment by arkpar** [comment]
Thank you for the PR. 
We need to be sure sure that `lz4_flex` and `lz` are 100% compatible. `lz4_flex` should be able to decompress existing data at least. I'm going to check that on a large enough real world database sometime soon.

File: admin/Cargo.toml [hunk 1]
```diff
@@ -4,6 +4,9 @@ version = "0.0.1"
 authors = ["Parity Technologies <admin@parity.io>"]
 publish = false
 edition = "2018"
+[features]
+default = []
```
[the remark this answers is not in the document]
Comment (afsec) [reply admin/Cargo.toml:8]: Due to jemalloc's nature as a binding of a C language library, compilation disruptions arise. An illustrative instance is the execution of the following command on an aarch64/arm64 architecture or a NixOS Linux distribution:
```
cargo build --release --target=$(arch -m)-unknown-linux-musl
```

**Review by afsec — COMMENTED** [review]
Upon the successful implementation of this commit, it becomes essential to employ the --no-default-features argument to ensure optimal portability for the administration tool. This can be accomplished by executing the command provided below:

```
cargo build --no-default-features --release --target=$(arch -m)-unknown-linux-musl
```

**Comment by afsec** [comment]
Thank you for your feedback. In order to ensure broad portability while simultaneously preserving the integrity of the original production-ready codebase, I have addressed the necessary changes, which are now reflected in this commit. I would greatly appreciate your review and feedback on these enhancements. Your insights are invaluable.

**Comment by arkpar** [comment]
Finally go to testing this, at the compatibility seems to be in order.

However I think adding all these features is an overkill and a maintenance burden. It would be enough to simply replace `lz` with `lz4_flex` and disable jemallocator for the msvc target.

**Comment by arkpar** [comment]
Let us know if you still interested in doing this. Otherwise we'll just patch it on our own.

**Comment by afsec** [comment]
Feel free to make any necessary changes. My goal was to make the library as simple and portable as possible by removing C/C++ language dependencies.

**Pull request closed without merging**